apps/pipelines/nodes/helpers.py

```diff
@@ -55,10 +55,26 @@ def _get_db_object(self):
             )
         return self._participant_data
 
-    def get(self):
-        return self._get_db_object().data
+    def get(self, include_global=True):
+        data = self._get_db_object().data
+        if include_global and hasattr(self.session, "participant") and hasattr(self.session.participant, "global_data"):
+            # Only add global data keys that don't already exist in participant data
+            for key, value in self.session.participant.global_data.items():
+                if key not in data:
+                    data[key] = value
+        return data
 
     def set(self, data):
         participant_data = self._get_db_object()
         participant_data.data = data
         participant_data.save(update_fields=["data"])
+        if hasattr(self.session, "participant") and hasattr(self.session.participant, "global_data"):
+            participant = self.session.participant
+            global_data_keys = list(participant.global_data.keys())
+            updated_fields = []
+            for key in global_data_keys:
+                if key in data and hasattr(participant, key):
+                    setattr(participant, key, data[key])
+                    updated_fields.append(key)
+            if updated_fields:
+                participant.save(update_fields=updated_fields)
```

Comment:
A nice option would be to put this on the participant object directly. Something like:

```python
participant.update_from_data(data)
```

That would encapsulate the 'global data' access and update within the participant itself. You could also use that method here: https://github.com/dimagi/open-chat-studio/blob/c3920a1639704d741fcf79f17ed14d4d57ea7ce4/apps/api/views.py#L154-L157